samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,395 @@
+    ///
+    /// # Input
+    /// ## bitsPerColor
+    /// Number of bits per color
+    /// ## register
+    /// register of qubits 
+    operation MeasureColoring (bitsPerColor : Int, register : Qubit[]) : Int[] {
+        let numVertices = Length(register) / bitsPerColor;
+        let colorPartitions = Partitioned(ConstantArray(numVertices - 1, bitsPerColor), register);
+        return ForEach(MeasureColor, colorPartitions);
+    } 
+
+    /// # Summary
+    /// N-bit color equality oracle (no extra qubits)
+    ///
+    /// # Input
+    /// ## color0
+    /// First color
+    /// ## color1
+    /// Second color
+    ///
+    /// # Output
+    /// target will be 1 if colors are the same
+    operation ApplyColorEqualityOracle (color0 : Qubit[], color1 : Qubit[], 
+        target : Qubit) : Unit is Adj+Ctl {
+        within {
+            for ((q0, q1) in Zip(color0, color1)) {
+                // compute XOR of q0 and q1 in place (storing it in q1)
+                CNOT(q0, q1);
+            }
+        } apply {
+            // if all XORs are 0, the bit strings are equal
+            (ControlledOnInt(0, X))(color1, target);
+        }
+    }
+
+    /// # Summary
+    /// Oracle for verifying vertex coloring, including color constraints from
+    /// non qubit vertices
+    ///
+    /// # Input
+    /// ## numVertices
+    /// The number of vertices in the graph
+    /// ## bitsPerColor
+    /// The bits per color e.g. 2 bits per color allows for 4 colors
+    /// ## edges
+    /// The array of (Vertex#,Vertex#) specifying the Vertices that can not be
+    /// the same color
+    /// ## startingColorConstraints
+    /// The array of (Vertex#,Color) specifying the dissallowed colors for vertices
+    ///
+    /// # Output
+    /// A unitary operation that applies `oracle` on the target register if the control 
+    /// register state corresponds to the bit mask `bits`.
+    ///
+    /// # Example
+    /// Consider the following 4x4 Sudoku puzzle 
+    ///     -----------------
+    ///     |   |   | 2 | 3 |
+    ///     -----------------
+    ///     |   |   | 0 | 1 |
+    ///     -----------------
+    ///     | 1 | 2 | 3 | 0 |
+    ///     -----------------
+    ///     | 3 | 0 | 1 | 2 |
+    ///     -----------------
+    ///  The challenge is to fill the empty squares with numbers 0 to 3
+    ///  that are unique in row, column and the top left 2x2 square
+    ///  This is a graph coloring problem where the colors are 0 to 3
+    ///  and the empty cells are the vertices. The vertices can be defined as   
+    ///     -----------------
+    ///     | 0 | 1 |   |   |
+    ///     -----------------
+    ///     | 2 | 3 |   |   |
+    ///     -----------------
+    ///     |   |   |   |   |
+    ///     -----------------
+    ///     |   |   |   |   |
+    ///     -----------------
+    /// The graph is
+    ///  0---1
+    ///  | X |
+    ///  1---2
+    /// i.e. every vertex is connected to each other
+    /// But we also require that 
```

Comment:
General writing style would suggest not starting a sentence with "but" if possible.